### ISSUE TITLE
wrong glibc version used for python and psycopg2

### ISSUE THREAD

**Issue by krimdomu** [cachix/devenv#891]
**Describe the bug**
I tried to build a devenv for one of our products. For this i need a python3 development environment and a postgresql database.
We are using psycopg2 to connect to the postgres database.
psycopg2 uses the libpq postgres client library (c-bindings).

The problem is, that postgres (and libpq) is linked against glibc-2.38 and python against glibc-2.37. This makes it impossible to use psycopg2.

```
(devenv) (venv) bash-5.2$ ldd $(which python) | grep libc.so
        libc.so.6 => /nix/store/flf14c3ibr83jsa070j25hg5gjapydhl-glibc-2.37-8/lib/libc.so.6 (0x00007f8b2233a000)
(devenv) (venv) bash-5.2$ ldd $(which psql) | grep libc.so
        libc.so.6 => /nix/store/qn3ggz5sf3hkjs2c797xf7nan3amdxmp-glibc-2.38-27/lib/libc.so.6 (0x00007fe8c7a41000)
(devenv) (venv) bash-5.2$ 
```
With `input` github:NixOS/nixpkgs/nixos-23.05 everything works as expected. But with `nixos-23.11` or `nixos-unstable` the bug is present.

You can test the python package as follow (inside `devenv shell`):

```
pip install psycopg2
echo "import psycopg2" | python -
```

You will get the following error:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jgehring/Projekte/inovex/selfservice/selfservice-backend-ng/.devenv/state/venv/lib/python3.11/site-packages/psycopg2/__init__.py", line 51, in <module>
    from psycopg2._psycopg import (                     # noqa
ImportError: /nix/store/flf14c3ibr83jsa070j25hg5gjapydhl-glibc-2.37-8/lib/libc.so.6: version `GLIBC_2.38' not found (required by /nix/store/mgn9k2dphgqxzlhhl6lih648pbrfzswp-devenv-profile/lib/libpq.so.5)
```

**To reproduce**

https://gist.github.com/krimdomu/064720c173217e5acf8012de44e32fc2

**Version**
 
devenv: 0.6.3

**Comment by txswdz** [comment]
I'm also experiencing this, though you can switch the branch in the flake without forking, by doing:
```
inputs.nixpkgs-python.inputs.nixpkgs.follows = "nixpkgs";
```
if you've got another input `nixpkgs` that is set to the right branch (i guess unstable or 23.11). Will changing this branch break something else?

**Comment by domenkozar** [comment]
This is fixed in 1.0.1
